src/Spago/Build.hs

```diff
@@ -331,10 +331,25 @@ getOutputPath
   :: Spago m
   => BuildOptions
   -> m (Maybe Sys.FilePath)
-getOutputPath _ = do
-
+getOutputPath buildOpts = do
   configPath <- asks globalConfigPath
-  PackageSet.findRootOutputPath (Text.unpack configPath)
+  outputPath <- PackageSet.findRootOutputPath (Text.unpack configPath)
+  case findOutputFlag (pursArgs buildOpts) of
+    Just path -> pure (Just path)
+    Nothing   ->
+      case shareOutput buildOpts of
+        NoShareOutput -> pure Nothing
+        ShareOutput   -> pure outputPath
+
+findOutputFlag :: [Purs.ExtraArg] -> Maybe Sys.FilePath
+findOutputFlag [] = Nothing
+findOutputFlag (_:[]) = Nothing
+findOutputFlag (x:y:xs)
+  = if (  Text.isInfixOf "-o" (Purs.unExtraArg x)
```

Comment:
Be careful, this test can match potentially more than you want:
1. `Text.isInfixOf "-o" "--output" == True`
2. `Text.isInfixOf "-o" "--other-thing" == True`

I'd suggest changing this to less liberal `"-o" == Purs.unExtraArg x`